File: client_leger/lib/classes/game-replay-commands/end_game_game_replay_command.dart
import 'package:client_leger/classes/game-replay-commands/game_replay_command.dart';
import 'package:client_leger/interfaces/dto/replay_event.dto.dart';
import 'package:client_leger/services/replay_service.dart';

class EndGameGameReplayCommand extends GameReplayCommand {
  final EndGameReplayDto eventDto;

  EndGameGameReplayCommand({required this.eventDto, required super.time});

  @override
  void action() {
    ReplayService.to.handleEndGame(eventDto.response);
  }
}

File: client/src/app/classes/end-game-game-replay-command/end-game-game-replay-command.ts
import { GameReplayCommand } from '@app/interfaces/game-replay-command';
import { ClassicModeService } from '@app/services/classic-mode/classic-mode.service';
import { EndGameResultDto } from '@common/model/dto/end-game-result';
import { ReplayEventDto } from '@common/model/events/replay.events';

export class EndGameGameReplayCommand implements GameReplayCommand {
    time: number;
    private classicModeService: ClassicModeService;
    private replayEventDto: ReplayEventDto;

    constructor(classicModeService: ClassicModeService, replayEventDto: ReplayEventDto) {
        this.time = replayEventDto.time;
        this.classicModeService = classicModeService;
        this.replayEventDto = replayEventDto;
    }

    action() {
        this.classicModeService.endGame(this.replayEventDto.response as EndGameResultDto);
    }
}
